utils/extract.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+
+# usage: extract.sh filename [destination]
+#   destination - directory to extract to. Will be created if non-existent. Leaving empty
+#     will result in extraction to the current working directory.
+
+# Requires:
+#   - tar
+#   - zip
+#   - 7z
+
+echoerr() {
+    echo "extract.sh:" "$@" >&2
+}
+
+
+# Preliminary checks
+if ! [ -f "$1" ]; then
+    echoerr "'$1' does not exist, aborting"
+    exit 1
+fi
+
+extension="${1#*.}"
+
+case "$extension" in
+    "zip")
+        cmd_needed="unzip"
+        ;;
+    "tar.gz")
```

Comment:
this also has an implicit dependency on a gzip implementation

Reply:
This comes standard with pretty much every Linux distro though, right?

Reply:
fixed